### PR TITLE
Complex numbers example

### LINKED ISSUE #448: Recipe: Complex Numbers
|  |  |
| ------------- | ------------- |
| Crates  | num |
| Section  | 16.1 Mathematics  |

Demonstrate using complex numbers (n+m√-1) with num crate

### PR DESCRIPTION
**Pull request by GuineaPiet**
Adds a Mathematics section with an example that shows how to create complex numbers.

fixes #448

### CODE REVIEW

File: src/SUMMARY.md [hunk 1]
```diff
@@ -53,3 +53,5 @@
     - [Making Requests](web/clients/requests.md)
     - [Calling a Web API](web/clients/apis.md)
     - [Downloads](web/clients/download.md)
+- [Mathematics](mathematics.md)
```
Comment (AndyGauge) [inline src/SUMMARY.md]: "Science"

File: src/SUMMARY.md [hunk 2]
```diff
@@ -53,3 +53,5 @@
...
+  - [Complex Numbers](mathematics/complex-numbers.md)
```
Comment (AndyGauge) [inline src/SUMMARY.md]: Mathematics

File: src/mathematics/complex-numbers.md [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+imaginary part of the complex number must be of the same type.
+
+```rust
+extern crate num;
+
+fn main() {
+    let complex_integer = num::complex::Complex::new(10, 20);
+    let complex_float = num::complex::Complex::new(10.1, 20.1);
+
+    println!("Complex integer: {}", complex_integer);
```
Comment (AndyGauge) [inline src/mathematics/complex-numbers.md]: So, creating complex numbers doesn't demonstrate their usefulness enough.   Can we perform some basic math with two complex numbers too?

**Comment by GuineaPiet** [comment]
Okay I made the changes as requested. I don't know in how much detail I'm supposed to go with the complex numbers 🤔 since explaining how they work or interact usually takes up an entire section in an advanced mathematics text book (and even then just to cover the basics). So I've shown how they can interact with each other and that the type itself has built-in functions if you want to use the number in mathematical functions.

**Comment by AndyGauge** [comment]
Love it

**Pull request merged**